src/rules/maxLineLengthRule.ts

```diff
@@ -29,12 +37,31 @@ export class Rule extends Lint.Rules.AbstractRule {
             Limiting the length of a line of code improves code readability.
             It also makes comparing code side-by-side easier and improves compatibility with
             various editors, IDEs, and diff viewers.`,
-        optionsDescription: "An integer indicating the max length of lines.",
+        optionsDescription: Lint.Utils.dedent`
+            An integer indicating the max length of lines, or a configuration object with two
+            properties: \`\"ignoreLongUrls\"\` and \`\"limit\"\`. If no configuration is provided, a
```

Comment:
Please add a note that urls are considered long if they are at least half of `limit` and that their length is subtracted from the total line length.